# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/nsRefreshDriver.cpp
# Commit: 2894ff06108e
# Full Hash: 2894ff06108e5fd9d16de8e5f80f9b04bae925f0
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2024-05-15 20:24:18
# Description:
#   Bug 1896024 - Ensure PerformanceHintSession is valid before reporting work duration. r=jesup,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D210042
# ==============================================================================

diff -r 09652347e814 -r 2894ff06108e layout/base/nsRefreshDriver.cpp
--- a/layout/base/nsRefreshDriver.cpp	Wed May 15 08:20:36 2024 +0000
+++ b/layout/base/nsRefreshDriver.cpp	Wed May 15 08:38:27 2024 +0000
@@ -831,11 +831,11 @@
     const TimeDuration previousRate = mVsyncRate;
     const TimeDuration rate = GetTimerRate();
 
-    hal::PerformanceHintSession* const performanceHintSession =
-        GetPerformanceHintSession();
-    if (performanceHintSession && rate != previousRate) {
-      performanceHintSession->UpdateTargetWorkDuration(
-          ContentChild::GetPerformanceHintTarget(rate));
+    if (rate != previousRate) {
+      if (auto* const performanceHintSession = GetPerformanceHintSession()) {
+        performanceHintSession->UpdateTargetWorkDuration(
+            ContentChild::GetPerformanceHintTarget(rate));
+      }
     }
 
     if (TimeDuration::FromMilliseconds(nsRefreshDriver::DefaultInterval()) >
@@ -862,7 +862,7 @@
 
     TimeStamp tickEnd = TimeStamp::Now();
 
-    if (performanceHintSession) {
+    if (auto* const performanceHintSession = GetPerformanceHintSession()) {
       performanceHintSession->ReportActualWorkDuration(tickEnd - tickStart);
     }
 
